src/Seabug/Metadata/Share.purs

```diff
@@ -0,0 +1,49 @@
+module Seabug.Metadata.Share
+  ( Share
+  , mkShare
+  , unShare
+  ) where
+
+import Prelude
+
+import Data.BigInt (BigInt)
+import Data.BigInt as BigInt
+import Data.Maybe (Maybe(Just, Nothing))
+import FromData (class FromData)
+import Metadata.FromMetadata (class FromMetadata)
+import Metadata.ToMetadata (class ToMetadata, toMetadata)
+import ToData (class ToData)
+import Types.Int (toBigInt) as Int
+import Types.PlutusData (PlutusData(Integer))
+import Types.TransactionMetadata (TransactionMetadatum(Int)) as Metadata
+
+-- | A number between 0 and 10000 (inclusive) representing percentage of the price.
+newtype Share = Share BigInt
+
+derive newtype instance ToData Share
+
+instance FromData Share where
+  fromData (Integer n) = BigInt.toInt n >>= mkShare
+  fromData _ = Nothing
+
+instance ToMetadata Share where
+  -- Must be safe when `Share` is built using `mkShare` smart constructor.
+  toMetadata = toMetadata <<< unShare
+
+instance FromMetadata Share where
+  fromMetadata (Metadata.Int n) =
+    BigInt.toInt (Int.toBigInt n) >>= mkShare
+  fromMetadata _ = Nothing
+
+instance Show Share where
+  show (Share share) = "(mkShare (" <> show share <> "))"
+
+derive instance Eq Share
+
+mkShare :: Int -> Maybe Share
+mkShare n
+  | n >= 0 && n <= 10000 = Just $ Share $ BigInt.fromInt n
```

Comment:
In Maks' paper the shares are defined to be between 0 and 1000, just checking if this is an intentional deviation? @samuelWilliams99 @t4ccer

Reply:
Sooo it's possible that I've made a typo when implementing it half a year ago and used `10000`. `10000` is now used everywhere

Reply:
https://github.com/mlabs-haskell/plutus-use-cases/blob/staging/mlabs/src/Mlabs/EfficientNFT/Token.hs#L147

Reply:
Hm, lets stick to `10000` and make a note of it, more precision is good, and 2dp on percentages is normal

Reply:
Deployed minting policy already uses `10000`, also it's better than `1000` for precision